server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -268,13 +271,30 @@ private void downloadToFS(String fileName, Path location, String primaryTerm) th
         fileTransferTracker.add(fileName, true);
     }
 
+    static public void verifyNoMultipleWriters(List<BlobMetadata> mdFiles) {
+        mdFiles.forEach(blobMetadata -> {
+            Map<String, List<String>> nodesByPrimaryTermAndGen = new HashMap<>();
+            Tuple<String, String> nodeIdByPrimaryTermAndGen = TranslogTransferMetadata.getNodeIdByPrimaryTermAndGen(
+                blobMetadata.toString()
+            );
+            if (nodeIdByPrimaryTermAndGen != null) {
+                nodesByPrimaryTermAndGen.computeIfAbsent(nodeIdByPrimaryTermAndGen.v1(), k -> new ArrayList<>());
+                nodesByPrimaryTermAndGen.get(nodeIdByPrimaryTermAndGen.v1()).add(nodeIdByPrimaryTermAndGen.v2());
+                if (nodesByPrimaryTermAndGen.get(nodeIdByPrimaryTermAndGen.v1()).size() > 1) {
+                    throw new IllegalStateException("Multiple metadata files having same primary term and generations detected");
```

Comment:
Can we enhance the code to also add the details about the primary term, generation along with nodes it has when the verification fails?

Reply:
Some of the code in verifyNoMultipleWriters for translog and segments looks to be duplicate. Possible to add this in utility and use same across both?

Reply:
done.

Reply:
done.